src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameControl.xaml.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Input;
+
+namespace Microsoft.CodeAnalysis.InlineRename.UI.SmartRename;
+
+/// <summary>
+/// Interaction logic for SmartRenameControl.xaml
+/// </summary>
+internal sealed partial class SmartRenameControl : UserControl
+{
+    private readonly SmartRenameViewModel _smartRenameViewModel;
+
+    public SmartRenameControl(SmartRenameViewModel viewModel)
+    {
+        this.DataContext = _smartRenameViewModel = viewModel;
+        InitializeComponent();
+    }
+
+    private void Suggestion_MouseDown(object sender, MouseButtonEventArgs e)
+    {
+        if (e.ClickCount == 1)
+        {
+            var identifierName = ((FrameworkElement)sender).Tag.ToString();
```

Comment:
how do we know that the sender is a framework eleemnt, or that the tag contains the info we need?

Reply:
By wpf definition https://learn.microsoft.com/en-us/dotnet/api/system.windows.frameworkelement.tag?view=windowsdesktop-7.0 
Then tag is directly binded to the display string in xaml
![image](https://github.com/dotnet/roslyn/assets/24360909/6923ba7e-2697-4bc8-9e8b-b5c4f6ac5b33)

![image](https://github.com/dotnet/roslyn/assets/24360909/1657bdb4-4cff-4969-9973-e0c453d1d586)

Reply:
Yea, this is safe in WPF as long as this method is used for mouse events.